zkvm/src/signature/signer.rs

```diff
@@ -165,28 +155,22 @@ impl PartyAwaitingShares {
         Signature { s, R: self.R }
     }
 
-    pub fn receive_shares(
-        self,
-        shares: Vec<Scalar>,
-        pubkeys: Vec<VerificationKey>,
-    ) -> Result<Signature, VMError> {
-        // Check that all shares are valid
-        for (i, s_i) in shares.iter().enumerate() {
-            let S_i = s_i * RISTRETTO_BASEPOINT_POINT;
-            let X_i = pubkeys[i];
-            let R_i = self.nonce_commitments[i].0;
-
-            // Make a_i = H(L, X_i)
-            let a_i = self.multikey.factor_for_key(&VerificationKey(X_i.0));
-
-            let X_i = X_i.0.decompress().ok_or(VMError::InvalidPoint)?;
+    pub fn receive_shares(self, shares: Vec<Scalar>) -> Result<Signature, VMError> {
+        // Create local copies of challenge and &multikey,
+        // since iterator takes ownership of `self` later.
```

Comment:
These are not local copies, these are moved-out fields, because compiler is not smart enough to capture them as moves in the map's closure.

how about:

```rust
// Move out self's fields because `self.c` inside `map`'s closure would
// lead to capturing `self` by reference, while we want
// to move `self.counterparties` out of it.
// See also RFC2229: https://github.com/rust-lang/rfcs/pull/2229
```

Reply:
thanks for clarification, will update!